### PR TITLE
Fix `view` in `view` method definition

### PR DESCRIPTION
**Pull request by akrmn**
https://github.com/digital-asset/ghc/pull/147

should fix https://github.com/digital-asset/daml/issues/15459

### CODE REVIEW

File: compiler/damlc/tests/daml-test-files/InterfaceMethodInMethod.daml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+-- Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+-- regression test for https://github.com/digital-asset/daml/issues/15459
```
Comment (akrmn) [inline compiler/damlc/tests/daml-test-files/InterfaceMethodInMethod.daml:4]: new regression test

File: compiler/damlc/tests/daml-test-files/InterfaceMethodInMethod.daml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  where
+    signatory p
+
+    interface instance I for T where
+      view = IV ()
+      im = succ
+
+    interface instance J for T where
+      -- using 'view' inside 'view' implementation
+      view = JV (view (toInterface @I this))
```
Comment (akrmn) [inline compiler/damlc/tests/daml-test-files/InterfaceMethodInMethod.daml:35]: the inner `view` refers to `DA.Internal.Interface.view : forall i v. HasInterfaceView i v => i -> v`

File: compiler/damlc/tests/daml-test-files/InterfaceMethodInMethod.daml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  with
+    p : Party
+    t : T
+  where
+    signatory p, t.p
+
+    interface instance I for U where
+      view = IV ()
+      -- using method 'im' inside method 'im' implementation
+      im = (2 *) . im (toInterface @I t)
```
Comment (akrmn) [inline compiler/damlc/tests/daml-test-files/InterfaceMethodInMethod.daml:47]: the inner `im` refers to `InterfaceMethodInMethod.m1 : I -> Int -> Int`

**Comment by dylant-da** [comment]
Great solution, thank you!

File: compiler/damlc/tests/daml-test-files/InterfaceMethodInMethod.daml (hunk 3 above)
Comment (akrmn) [reply compiler/damlc/tests/daml-test-files/InterfaceMethodInMethod.daml:47]: very good point, thank you Dylan!

**Pull request merged**